File: common/src/main/java/me/kvalbrus/multibans/common/storage/DataProviderSettings.java
package me.kvalbrus.multibans.common.storage;

import com.moandjiezana.toml.Toml;
import com.moandjiezana.toml.TomlWriter;
import java.io.File;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;
import java.util.Properties;
import lombok.Getter;
import me.kvalbrus.multibans.common.managers.PluginManager;
import org.jetbrains.annotations.NotNull;

import static me.kvalbrus.multibans.common.storage.DataProviderType.MY_SQL;

public class DataProviderSettings {

    private static final String PATH = "data_provider.toml";

    private final PluginManager pluginManager;

    @Getter
    private DataProviderType type = MY_SQL;

    @Getter
    private Properties properties;

    public DataProviderSettings(@NotNull PluginManager pluginManager) {
        this.pluginManager = pluginManager;
        this.properties = new Properties();
    }

    @NotNull
    public synchronized DataProviderSettings load() {
        File file = new File(this.pluginManager.getDataFolder(), PATH);

        if (!file.exists()) {
            try {
                file.createNewFile();
            } catch (IOException exception) {
                return this;
            }
        }

        Toml toml = new Toml().read(file);
        TomlWriter tomlWriter = new TomlWriter.Builder().build();

        Map<String, Object> map = new HashMap<>();


        DataProviderType type = DataProviderType.getType(toml.getString("name"));
        if (type != null) {
            this.type = type;
        }

        this.properties.setProperty("dataSource.databaseName", toml.getString("databaseName") != null ?
            toml.getString("databaseName") : "MultiPunishments");
        this.properties.setProperty("dataSource.serverName", toml.getString("address") != null ?
            toml.getString("address") : "localhost");
        long port = (toml.getLong("port") != null && toml.getLong("port") > 0) ? toml.getLong("port") : 3306L;
        this.properties.setProperty("dataSource.portNumber", String.valueOf(port ));
        this.properties.setProperty("dataSource.user", toml.getString("user") != null ?
            toml.getString("user") : "");
        this.properties.setProperty("dataSource.password", toml.getString("password") != null ?
            toml.getString("password") : "");

        map.put("name", this.type.getName());
        map.put("databaseName", this.properties.getProperty("dataSource.databaseName"));
        map.put("address", this.properties.getProperty("dataSource.serverName"));
        map.put("port", port);
        map.put("user", this.properties.getProperty("dataSource.user"));
        map.put("password", this.properties.getProperty("dataSource.password"));

        try {
            tomlWriter.write(map, file);
        } catch (IOException exception) {
            return this;
        }

        return this;
    }
}

File: common/src/main/java/me/kvalbrus/multibans/common/storage/DataProviderSettings.kt
package me.kvalbrus.multibans.common.storage

import com.moandjiezana.toml.Toml
import com.moandjiezana.toml.TomlWriter
import lombok.Getter
import me.kvalbrus.multibans.common.managers.PluginManager
import java.io.File
import java.io.IOException
import java.util.*

class DataProviderSettings(private val pluginManager: PluginManager) {

    var type = DataProviderType.MY_SQL
    val properties: Properties

    init {
        properties = Properties()
    }

    @Synchronized
    fun load(): DataProviderSettings {
        val file = File(pluginManager.dataFolder, PATH)
        if (!file.exists()) {
            try {
                file.createNewFile()
            } catch (exception: IOException) {
                return this
            }
        }
        val toml = Toml().read(file)
        val tomlWriter = TomlWriter.Builder().build()
        val map: MutableMap<String, Any> = HashMap()
        val type = DataProviderType.getType(toml.getString("name"))
        if (type != null) {
            this.type = type
        }
        properties.setProperty(
            "dataSource.databaseName",
            if (toml.getString("databaseName") != null) toml.getString("databaseName") else "MultiPunishments"
        )
        properties.setProperty(
            "dataSource.serverName",
            if (toml.getString("address") != null) toml.getString("address") else "localhost"
        )
        val port =
            if (toml.getLong("port") != null && toml.getLong("port") > 0) toml.getLong("port") else 3306L
        properties.setProperty("dataSource.portNumber", port.toString())
        properties.setProperty(
            "dataSource.user",
            if (toml.getString("user") != null) toml.getString("user") else ""
        )
        properties.setProperty(
            "dataSource.password",
            if (toml.getString("password") != null) toml.getString("password") else ""
        )
        map["name"] = this.type.name
        map["databaseName"] = properties.getProperty("dataSource.databaseName")
        map["address"] = properties.getProperty("dataSource.serverName")
        map["port"] = port
        map["user"] = properties.getProperty("dataSource.user")
        map["password"] = properties.getProperty("dataSource.password")
        try {
            tomlWriter.write(map, file)
        } catch (exception: IOException) {
            return this
        }
        return this
    }

    companion object {
        private const val PATH = "data_provider.toml"
    }
}